os_brick/initiator/connectors/lightos.py

```diff
@@ -187,66 +188,87 @@ def get_volume_paths(self, connection_properties):
         path = connection_properties['device_path']
         return [path]
 
-    @utils.trace
-    def _get_device_by_uuid(self, uuid):
-        lnk_path = "%s%s" % ("/dev/disk/by-id/nvme-uuid.", uuid)
+    def _check_device_exists_using_dev_lnk(uuid):
+        lnk_path = f"/dev/disk/by-id/nvme-uuid.{uuid}"
+        if os.path.exists(lnk_path):
+            devname = os.path.realpath(lnk_path)
+        if devname.startswith("/dev/nvme"):
+            devname = devname.strip()
+            LOG.info("LIGHTOS: devpath %s detected for uuid %s",
+                     devname, uuid)
+            return devname
+        return None
+
+    def _check_device_exists_reading_block_class(uuid):
         file_path = "/sys/class/block/*/wwid"
         wwid = "uuid." + uuid
-        start = time.time()
-        while time.time() < start + self.WAIT_DEVICE_TIMEOUT:
+        for match_path in glob.glob(file_path):
+            try:
+                with open(match_path, "r") as f:
+                    match_wwid = f.readline()
+            except Exception:
+                LOG.warning("LIGHTOS: Failed to read file %s",
```

Comment:
it would be better to log the exception cause here, so we know why we failed to read. Also, if this is expected to happen, we could spam the logs. I would probably make it debug.